test/test_suite.cpp

```diff
@@ -1079,18 +1079,36 @@ TEST_CASE("float", "[]" ) {
   char buffer[100];
 
   // test special-case floats using math.h macros
+  test::sprintf(buffer, "%8f", nan(""));
+  REQUIRE(!strcmp(buffer, "     nan"));
+
+  test::sprintf(buffer, "%8f", (double)nanf(""));
+  REQUIRE(!strcmp(buffer, "     nan"));
+
   test::sprintf(buffer, "%8f", NAN);
   REQUIRE(!strcmp(buffer, "     nan"));
 
   test::sprintf(buffer, "%8f", INFINITY);
   REQUIRE(!strcmp(buffer, "     inf"));
 
-  test::sprintf(buffer, "%-8f", -INFINITY);
+  test::sprintf(buffer, "%-8f", (double)-INFINITY);

```

Comment:
Same for all these:
https://en.cppreference.com/w/cpp/types/numeric_limits/infinity

Reply:
Got it.